File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const countMap = new Map();

    for (let t of tangerine) {
        countMap.set(t, (countMap.get(t) || 0) + 1);
    }

    const sortedCounts = Array.from(countMap.values()).sort((a, b) => b - a);

    let total = 0;
    let kind = 0;

    for (let count of sortedCounts) {
        total += count;
        kind++;
        if (total >= k) break;
    }

    return kind;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int t : tangerine){
            map.put(t, map.getOrDefault(t, 0) + 1);
        }
        
        List<Integer> keys = new ArrayList<>(map.keySet());
        keys.sort((a, b) -> (map.get(b) - map.get(a)));
        
        int tan = 0;
        for(int i = 0; i < keys.size(); i++){
            if(tan >= k){
                return i;
            }else{
                tan += map.get(keys.get(i));
            }
        }
        return keys.size();
    }
}